From 3f96c35a50c8f1da7d6b3ba0f38dffafe93cdf03 Mon Sep 17 00:00:00 2001
From: Evan Wang <xswang@marvell.com>
Date: Fri, 11 Aug 2017 12:17:58 +0800
Subject: [PATCH 1181/1345] fix: pci: aardvark: skip PCIe reset if no valid
 gpio set

commit  281fb64c6d46d1e15fef14187ae3535281ea58d1 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

There is an issue that if there is no valid PCIe reset
GPIO set in DTS, the PCIe driver will have a warnning
crash appear when clod boot.
The patch fix the issue via skipping the PCIe reset
if there is no valid GPIO setting in PCIe DTS node.

Change-Id: I0ee6389ff386f6bf242ed3d74c720a980c4e8669
Signed-off-by: Evan Wang <xswang@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/42890
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Victor Gu <xigu@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/pci/host/pci-aardvark.c |   22 +++++++++++-----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/drivers/pci/host/pci-aardvark.c b/drivers/pci/host/pci-aardvark.c
index 223e811..377f8a0 100644
--- a/drivers/pci/host/pci-aardvark.c
+++ b/drivers/pci/host/pci-aardvark.c
@@ -978,18 +978,18 @@ static int advk_pcie_probe(struct platform_device *pdev)
 		return PTR_ERR(pcie->clk);
 	}
 
-	/* Config reset gpio for pcie */
+	/* Config reset gpio for pcie if there is valid gpio setting in DTS */
 	reset_gpio = of_get_named_gpio_flags(dn, "reset-gpios", 0, &pcie->flags);
-	if (reset_gpio != -EPROBE_DEFER) {
-		if (gpio_is_valid(reset_gpio)) {
-			pcie->reset_gpio = gpio_to_desc(reset_gpio);
-			ret = advk_pcie_clk_enable_then_reset(pcie);
-			if (ret)
-				return ret;
-
-			/* continue init flow after pcie reset */
-			goto after_pcie_reset;
-		}
+	if (gpio_is_valid(reset_gpio)) {
+		pcie->reset_gpio = gpio_to_desc(reset_gpio);
+		ret = advk_pcie_clk_enable_then_reset(pcie);
+		if (ret)
+			return ret;
+
+		/* continue init flow after pcie reset */
+		goto after_pcie_reset;
+	} else if (reset_gpio == -EPROBE_DEFER) {
+		return -EPROBE_DEFER;
 	}
 
 	ret = clk_prepare_enable(pcie->clk);
-- 
1.7.9.5

